components/bl-dashlet/src/helpers.js

```diff
@@ -124,3 +124,98 @@ export const getPosition = (ref, coords) => {
     ),
   };
 };
+
+export const useContextMenuPositionHandler = (contextMenuRef, isMenuOpen) => {
+  const [sides, setSides] = useState({ left: false, top: false });
+  const [isChecked, setIsChecked] = useState(false);
+
+  useEffect(() => {
+    if (contextMenuRef.current && isMenuOpen) {
+      contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+
+      window.addEventListener('resize', () => {
+        contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+      });
+
+    } else {
+      setSides({ left: false, top: false });
+      setIsChecked(false);
+    }
+
+    return () => {
+      window.removeEventListener('resize', () => {
+        contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+      });
+    };
+  }, [contextMenuRef, isMenuOpen]);
+
+  return { sides, isChecked };
+};
+
+const contextMenuPosition = (contextMenuRef, isMenuOpen, sides, setIsChecked, setSides) => {
+  const { bottom, right } = contextMenuRef.current.getBoundingClientRect();
+
+  if (right > window.innerWidth) {
+    setSides({ ...sides, left: true });
+  }
+
+  if (bottom > window.innerHeight) {
+    setSides({ ...sides, top: true });
+  }
```

Comment:
I know there will be a batching. But in any case i preffer to use one setter if it possible

```
  if (bottom > window.innerHeight || right > window.innerWidth) {
    setSides({ left: right > window.innerWidth, top: bottom > window.innerHeight});
  }
```

Just FYI, setter should be removed in any case.